cypress/e2e/stepDefinition/commonSteps.js

```diff
@@ -0,0 +1,118 @@
+import { Then, And, When } from 'cypress-cucumber-preprocessor/steps';
+import FilterPage from '../pages/filterPage';
+import HomePage from '../pages/homePage';
+
+Then('user verifies {string} link contains {string} text', (link, text) => {
+  cy.get('footer').scrollIntoView();
+  cy.get(FilterPage.links[link]).contains(text);
+});
+
+Then('user verifies demo data {string} is visible', (text) => {
+  cy.get(FilterPage.links['DEMO_DATA']).contains(text);
+});
+
+When('user clicks on {string} button', (button) => {
+  const element = cy.get(FilterPage.buttons[button]);
+  element.click();
+});
+
+When('user clicks on side bar filter button', () => {
+  const element = cy.get(HomePage.buttons['SIDE_BAR_FILTER']);
+  element.click();
+});
+
+And('user see {int} number of filtered launches', (expectedQuantities) => {
+  expect(expectedQuantities).within(2, 50);
+});
+
+Then('user clicks on more condition button', () => {
+  const element = cy.get(FilterPage.buttons['MORE']).contains('More');
+  element.click();
+});
+
+And('user selects {string} condition', (condition) => {
+  const element = cy.get(FilterPage.buttons['MORE']).contains(condition);
+  element.click();
+});
+
+Then(
+  'user verifies filter label {string} has a value {string}',
+  (label, text) => {
+    const element = cy.get(FilterPage.labels[label]).contains(text);
+    element.should('be.visible');
+  }
+);
+
+And(
+  'user sets value {string} into {string} input field',
+  (textInput, field) => {
+    const element = FilterPage.fields.INPUT(field);
+    if (field === 'Enter filter name') {
+      const number = Math.floor(Math.random() * 1000);
+      textInput = textInput + ' ' + number;
+    }
+
+    cy.get(element).should('be.visible').clear().type(textInput);
+    cy.wait(5000);
```

Comment:
We should avoid using `wait` in Cypress. This should be the last case to use. See more about it:
https://docs.cypress.io/guides/references/best-practices#Unnecessary-Waiting

Reply:
understand. Added wait to avoid flakiness - after setting the value into search control the filters table below is not rendered immediately. so the next step with assertion catches the wrong line.